system_metrics_collector/src/system_metrics_collector/linux_cpu_measurement_node.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SYSTEM_METRICS_COLLECTOR__LINUX_CPU_MEASUREMENT_NODE_HPP_
+#define SYSTEM_METRICS_COLLECTOR__LINUX_CPU_MEASUREMENT_NODE_HPP_
+
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "../../src/system_metrics_collector/periodic_measurement_node.hpp"
+#include "../../src/system_metrics_collector/cpu_data.hpp"
+
+/**
+ * Node that periodically calculates the % of active CPU by
+ * reading /proc/stat.
+ */
+class LinuxCpuMeasurementNode : public PeriodicMeasurementNode
+{
+public:
+  static constexpr const char PROC_STAT_FILE[] = "/proc/stat";
+  static constexpr const char CPU_LABEL[] = "cpu";
+  static constexpr const size_t CPU_LABEL_LENGTH = 3;
```

Comment:
Does this need to be public? If no, could we move this to the cpp file?

Generally, it is good practice to avoid exposing things we don't need to.